tests/integration/test_akafka_testutils.py

```diff
@@ -136,15 +136,17 @@ async def test_clear_topics_specific(kafka: KafkaFixture):
 
     # make sure messages are consumed again
     records = []
-    while True:
-        prefetched = await consumer.getmany(timeout_ms=5000)
-        if not prefetched:
+    while (fetches := 0) < 10:
+        # use limit in case of changes that cause unintended infinite loop here
+        fetches += 1
+        prefetched = await consumer.getmany(timeout_ms=500)
+        if prefetched:
+            records.extend(next(iter(prefetched.values())))
+        if len(records) >= 2:
             break
```

Comment:
We could put something like `wait asyncio.sleep(0.05)` here at the very end of the loop.